### PR TITLE
fix: Fix tests which were failing

### PR DESCRIPTION
**Pull request by bharath-123**
This PR fixes all the failing tests in the astria-geth repo apart from the [`TestReimportMirroredState` ](https://github.com/astriaorg/astria-geth/blob/f5a95fe13d581ae1ec67caeaae6efe5338399a3e/consensus/clique/clique_test.go#L44) (More details at the end)

The main reasons for the tests failing were:
1. https://github.com/astriaorg/astria-geth/pull/5 which transfers the basefee to the fee recipient rather than burning it. This caused all the tests which expect a certain state root to fail since they assume that the basefee would be burnt. It also failed tests which check for the balance of the fee recipient which fails because we also have to account for the base fee. To fix them, we replaced the tests with the expected state root and updated the tests with the expected balance of the fee recipient. 
2. Picking txs from the `AstriaOrdered` rather than from the geth mempool. Lot of tests place the txs in the Geth mempool and they end up failing as we do not pick txx from their. To fix this, we had to add these txs to the `AstriaOrdered` mempool instead

This PR also adds a github action to run the tests.

`TestReimportMirroredState` is very tricky to fix since the fee recipient while building the block and while verifying the block before inserting onchain is different. This is because while verifying the block, we pick the fee recipient from the consensus engine's author which in this case is the signer of the block. It's tricker to fix it and we have commented it out for now.

### CODE REVIEW

File: metrics/influxdb/influxdb_test.go [hunk 1]
```diff
@@ -37,6 +38,10 @@ func TestMain(m *testing.M) {
 }
 
 func TestExampleV1(t *testing.T) {
+	if runtime.GOARCH == "arm64" {
+		t.Skip("test skipped on ARM64 due to floating point precision differences")
+	}
+
```
Comment (bharath-123) [inline metrics/influxdb/influxdb_test.go:44]: This is picked up from the geth client. they skip these tests on arm64

File: miner/worker_test.go [hunk 2]
```diff
@@ -1,501 +0,0 @@
-// Copyright 2018 The go-ethereum Authors
```
Comment (bharath-123) [inline miner/worker_test.go:1]: This file is removed from the OG geth repo. It's better if we remove it too.

File: tests/block_test.go [hunk 3]
```diff
@@ -56,17 +55,20 @@ func TestBlockchain(t *testing.T) {
...
+//func TestExecutionSpec(t *testing.T) {
+//	if !common.FileExist(executionSpecDir) {
+//		t.Skipf("directory %s does not exist", executionSpecDir)
+//	}
+//	bt := new(testMatcher)
+//
+//	bt.walk(t, executionSpecDir, func(t *testing.T, name string, test *BlockTest) {
+//		execBlockTest(t, bt, test)
+//	})
+//}
```
Comment (bharath-123) [inline tests/block_test.go:71]: Need to debate whether to keep these tests or not.

File: .github/workflows/run-tests.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.21.4
+      - name: Run tests
+        run: go test -short ./...
+        env:
+          GOOS: linux
```
Comment (joroshiba) [inline .github/workflows/run-tests.yml:25]: nit: new line at end of file

**Review by joroshiba — APPROVED** [review]

**Pull request merged**